### PR TITLE
Job Improvement Project - Genetic

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: #2571

* **Server Mode**: Renewal

* **Description of Pull Request**: 
  * Updated Homunculus skills from kRO maintenance: http://ro.gnjoy.com/news/event/View.asp?Seq=784&kind=A&curpage=1
  * Updated more Homunculus skills and Genetic skills from kRO maintenance: http://ro.gnjoy.com/news/update/View.asp?seq=246&curpage=1

### CODE REVIEW

**Comment by admkakaroto** [comment]
the experience gain for the homunculi is not working.

**Comment by aleos89** [comment]
In what way? They aren't gaining any experience? What server mode are you using?

**Comment by admkakaroto** [comment]
@aleos89 nop, any exp, server in RE mode.

**Comment by aleos89** [comment]
The homun is gaining experience, it's just such a small value it's not very noticeable. I adjusted the formula since the translation says it takes the level into account, but not sure how it does.

**Comment by admkakaroto** [comment]
ezgif-3-30abd82423

I can not understand, it does not gain anything from exp ...

**Comment by AzarthMZintos** [comment]
the exp of the monster the homunculus kills by him self go to the owern and 10% of that go to the homunculus, not all exp of the mob the owener kill in this image the owner is killing not the homunculus

**Comment by admkakaroto** [comment]
I know this, the homunculus should not receive 10% of the experience the master receives by killing monsters?

**Comment by AzarthMZintos** [comment]
try to just let the homunclus kill or do mora damage than you

**Comment by AzarthMZintos** [comment]
you said that base on what?

**Comment by cydh** [comment]
^ what he mentioned is still from #2571, 
> 10% of the experience obtained by the Genetic is given to the homunculus adjusted to its base level.

**Comment by AzarthMZintos** [comment]
Ok

**Comment by AzarthMZintos** [comment]
"All the experience when the homunculus kills a target is given to the Genetic (including JEXP)."
will this be a custom option? if no can you make it?

**Comment by spinzaku** [comment]
hey @aleos89  i think this broke the homunculus autofeed

**Comment by Badarosk0** [comment]
Autofeed is not working.

**Comment by Badarosk0** [comment]
I understand, autofeed was running on the rA for some time, but after upgrades it stopped working.

**Comment by admkakaroto** [comment]
It works perfectly for me.

**Comment by ecdarreola** [comment]
After the latest revision, homun not receiving exp no matter what level of the master.

**Comment by aleos89** [comment]
Actually, I completely forgot that the homunculus always gains EXP and the master. That change above breaks it. I will fix it.

**Comment by Badarosk0** [comment]
> Actually, I completely forgot that the homunculus always gains EXP and the master. That change above breaks it. I will fix it.

With this change will generate crash over time.

**Comment by Zarbony** [comment]
Got an Error, always the Player gain EXP, a Homunculus will gain EXP to but if the Player hasnt any Homunculus, this one wont Stop.
This causes an Map Server Crash.

nullpo info ho

**Comment by ecdarreola** [comment]
Is it normal that a player can receive 2 sets of exp /w homunculus? 1 is from the player and 1 from the homunculus itself?

image

**Comment by aleos89** [comment]
They are both displayed at the same time from the same mob's death?

**Comment by ecdarreola** [comment]
@aleos89 : yes correct on mob_dead or mob's death it displayed at the same time. No delay.

File: db/re/exp_homun.txt [hunk 1]
```diff
@@ -1,151 +1,176 @@
...
+50572500
+53680000
+57410000
+61887500
+70842500
+75320000
+80692500
+87140000
+94875000
+0
```
[the remark this answers is not in the document]
Comment (aleos89) [reply db/re/exp_homun.txt]: The previous database was the same way, where level 150 had a value of 0.

**Comment by LordWhiplash** [comment]
Hell Expansion is not working properly for area acid bomb, the damage is very low compared to the acid bomb damage

**Comment by Feelmeone** [comment]
Hell plant must be removed attack motion.
77YYTPVA0SQO16BPUAM5

**Pull request merged**